### DISCUSSION TITLE
Example on how to disconnect a bot from a voice channel after it is done with playing audio?

### DISCUSSION THREAD

**Discussion by FlyingThunder** [Rapptz/discord.py#7223] [Q&A]
Ive been looking for a clear example of that for a while.

So far, ive been using that:

```
         while vc.is_playing() == False:
             for x in bot.voice_clients:
                 if (x.guild == message.guild):
                     await x.disconnect()
```

but with time this lagged my bot very hard while playing audio.
I couldnt find a more efficient way of checking that, examples very appreciated

**Comment by FlyingThunder** [comment]
found a solution, hopefully itll work long time:
```py
from asyncio import sleep
while voice_client.is_playing():
  await sleep(1)
await voice_client.disconnect()
```

**Comment by GnomedDev** [comment]
Just use the after kwarg of vc.play. https://discordpy.readthedocs.io/en/stable/api.html#discord.VoiceClient.play
